src/Scaffolding/VS.Web.CG.Mvc/BlazorIdentity/BlazorIdentityGenerator.cs

```diff
@@ -0,0 +1,544 @@
+                        blazorIdentityModel.UserClassNamespace = IdentityHelper.GetNamespaceFromTypeName(commandlineModel.UserClass)
+                            ?? defaultDbContextNamespace;
+                    }
+                }
+            }
+
+            var applicationUserModel = new IdentityApplicationUserModel
+            {
+                UserClassName = blazorIdentityModel.UserClassName,
+                UserClassNamespace = blazorIdentityModel.UserClassNamespace
+            };
+
+            if (createUserClass)
+            {
+                ExecuteApplicationUserTemplate(applicationUserModel);
+            }
+
+            if (createDbContext)
+            {
+                var dbContextModel = new IdentityDbContextModel
+                {
+                    DbContextName = blazorIdentityModel.DbContextName,
+                    DbContextNamespace = blazorIdentityModel.DbContextNamespace,
+                    UserClassModel = applicationUserModel
+                };
+
+                ExecuteDbContextTemplate(dbContextModel);
+                var connectionStringsWriter = new ConnectionStringsWriter(AppInfo, FileSystem);
+                connectionStringsWriter.AddConnectionString("DefaultConnection", dbContextModel.DbContextName, commandlineModel.DatabaseProvider);
+            }
+
+            //gather blazor identity files to generate
+            blazorIdentityModel.FilesToGenerate = new List<string>();
+            if (!string.IsNullOrEmpty(commandlineModel.Files))
+            {
+                blazorIdentityModel.FilesToGenerate = commandlineModel.Files.Split(IdentityHelper.SemicolonSeparator, StringSplitOptions.RemoveEmptyEntries).ToList();
+            }
+            else if (!string.IsNullOrEmpty(commandlineModel.ExcludeFiles))
+            {
+                IEnumerable<string> excludedFiles = commandlineModel.ExcludeFiles.Split(IdentityHelper.SemicolonSeparator, StringSplitOptions.RemoveEmptyEntries).Select(p => p.Trim()).ToList();
+                //validate excluded files
+                var errors = new List<string>();
+                var invalidFiles = excludedFiles.Where(f => !AllBlazorIdentityFiles.ContainsKey(f));
+                if (invalidFiles.Any())
+                {
+                    errors.Add(MessageStrings.InvalidFilesListMessage);
+                    errors.AddRange(invalidFiles);
+                }
+
+                if (errors.Count != 0)
+                {
+                    throw new InvalidOperationException(string.Join(Environment.NewLine, errors));
+                }
+
+                //get files to overwrite
+                blazorIdentityModel.FilesToGenerate = AllBlazorIdentityFiles.Keys.Except(excludedFiles).ToList();
+            }
+
+            if (blazorIdentityModel.FilesToGenerate.Count == 0)
+            {
+                blazorIdentityModel.FilesToGenerate = AllBlazorIdentityFiles.Keys.ToList();
```

Comment:
This might impact the answer to my last comment about line 265, but is the intention here to do this even if one of the two conditions above executed and the result of them was an empty set? Or only if neither condition was true?

Reply:
If neither condition is true, then we just go ahead scaffold all identity files is the intent. If we end up with an empty set from the 2 conditions, it's ok to scaffold I presume. I think that new `FilesToGenerate` list is probably unnecessary so will just add a final else statement. tyty